lib/src/HttpRequestImpl.h

```diff
@@ -554,6 +563,26 @@ class HttpRequestImpl : public HttpRequest
         return keepAlive_;
     }
 
+    bool connected() const noexcept override
+    {
+        if (auto conn = connPtr_.lock())
+        {
+            auto loopPtr = conn->getLoop();
+            if (loopPtr->isInLoopThread())
+            {
+                return conn->connected();
+            }
+            else
+            {
+                std::promise<bool> flag;
+                loopPtr->queueInLoop(
+                    [&] { flag.set_value(conn->connected()); });
+                return flag.get_future().get();
```

Comment:
I think there is a bit of overprotection here; just returning conn->connected() directly is enough. Even if protection is implemented here for cross-thread calls, the caller will still face race conditions when getting the result. It’s better to give the choice to the user. Those who need strict consistency can use queueInLoop to ensure it.

Reply:
ok,I'll change it tomorrow btw ,the user may not actually notice a race conditions，Whether to consider using atomic to store connection state

Reply:
done